auth/gcloud/aio/auth/session.py

```diff
@@ -335,7 +338,9 @@ async def get(
             timeout: float = 10,
             params: Optional[Mapping[str, Union[int, str]]] = None,
             stream: bool = False,
+            auto_decompress: Optional[bool] = None,
         ) -> Response:
+            del auto_decompress  # Not used in SyncSession
```

Comment:
We're generally very cautious on introducing functionality which varies between the sync and aio versions of this library. It looks like both `requests` and `aiohttp` automatically decompress by default, but only the latter lets you disable the functionality.

We should do one of the following:
* remove this flag, eg. such that both libraries act the same and always decompress
* throw an explicit error to users if `auto_decompress != True` in the sync version of the library

I am leaning quite heavily toward the former of those options, but if you've got a strong use-case for needing the libraries to act differently, we could consider the latter. Thoughts?

Reply:
From a client's standpoint, their code doesn't need to change, because of the `if not auto_decompress and BUILD_GCLOUD_REST:` clause introduced in `_download`. This let's you access the compressed bytes when using the `requests` lib. I verified with the integration tests and I also manually and it works as expected. By passing `auto_decompress` to `blob.download()` you are just indicating to our lib that you want the compressed file.

I think this is a totally valid use case, you could very well want to grab a gzip file from GCS and store it zipped somewhere else for serving or later processing. Automatically decompressing just to compress it again is wasteful.

Reply:
Aha, I had missed that `requests` avoids decompressing if you do a streaming download, in which case you are entirely correct that we can coerce the behavior into being the same, so this looks good as-is.